Api impl

Labyc opened the pull request:
[+] Person model
[...] split passport types
[+] Person endpoints controller
[+] ginignore

TerKaCe committed:
[+] Invalid data error response
[...] responses
[...] formats
[+] internal passport ID
[+] internal Person ID
[...] swap person to personID in passport entity
[...] passport fields
[...] patch methods
[+] put method for passport
[+] search passport by person ID
[-] search passport by person

TerKaCe committed:
[...] query parameters for get,patch,put requests

TerKaCe committed:
[...] split in and out entities
[...] minor fixes

TerKaCe committed:
[-] put passport
[+] put person

TerKaCe committed:
[...] endpoints
[...] errors
[...] models

TerKaCe committed:
[...] fix syntax

TerKaCe committed:
[+] Person model
[...] split passport types
[+] Person endpoints controller
[+] ginignore

TerKaCe committed:
[+] examples of validation
[+] example exception+check
[+] personDTO

TerKaCe committed:
[+] person endpoint implementation
[...] refactor validation
[...] refactor person model and it's DTOs
[...] error responce may contain multiple errors

TerKaCe committed:
[-] javax.persistence
[...] refactor Person model and DTOs

TerKaCe committed:
unworkable commit, just to save before vipe

Commit:
[...] unworkable commit, to move to new PC

mih-kopylov commented on src/main/java/app/bootstrap/BootStrap.java:
```diff
@@ -1,28 +1,31 @@
 
 
 @Component
 public class BootStrap implements ApplicationListener<ContextRefreshedEvent> {
```
it seems there's no need in this component

mih-kopylov commented on src/main/java/app/ErrorResponse.java:
```diff
@@ -0,0 +1,37 @@
+
+@NoArgsConstructor
+@Getter
+public class ErrorResponse{
```
this seems like a controller level response DTO, therefore should be in controller's package

mih-kopylov commented on src/main/java/app/models/passport/passportRFForeign/PassportRFForeign.java:
```diff
@@ -0,0 +1,16 @@
+public class PassportRFForeign extends BasePassport {
+    @NotNull
+    private Date expirationDate;
+    @Digits(integer = 8, fraction = 0, message = "Invalid Passport number")
```
This is a controller level validation, but the class is in `model` package.
Consider separating models and controller's DTOs (requests and responses)

mih-kopylov commented on src/main/java/app/models/person/Person.java:
```diff
@@ -0,0 +1,43 @@
+public class Person {
+    @Setter
+    UUID id;
+    String name;
```
The models are safer to be immutable. Consider making them `private final` and using lombok's `@With` or analog

mih-kopylov commented on src/main/java/app/models/person/Person.java:
```diff
@@ -0,0 +1,43 @@
+    String surname;
+    String patronymic;
+    String placeOfBirth;
+    Date dateOfBirth;
```
`java.util.Date` is an old way of managing dates.
Consisder using new `java.time` package instead.

mih-kopylov commented on src/main/java/app/models/personRegistration/PersonRegistration.java:
```diff
@@ -1,4 +1,4 @@
-package app.models;
+package app.models.personRegistration;
```
Package name should be in lower case accodring to java conversions. It's also better to have `person.registration` package instead of a single `personregistration`

TerKaCe committed:
[...] migrate to Spring-boot 3
[+] Swagger doc autogeneration
[-] Data storage and replace with repository interface and impl
[...] rename PersonNotFoundException to EntityNotFoundException
[-] PassportNotFoundException
[...] UUID to String of UUID
[+] ID to ErrorResponse
[...] enhancement of ErrorResponse
[+] handle HttpMessageNotReadableException
[...] Fix pattern for Name, Surname, Patronymic, Place of Birth
[...] Patronymic now can be set empty via patch with ""
[+] Now it can be packed to JAR

Labyc committed:
Merge pull request #3 from Labyc/temp

Person API impl

TerKaCe committed:
[+] find person feature
[+] PersonProcessor and PersonRepository specific interface for person specific methods
[...] Exception handlers moved to interface over controllers

TerKaCe committed:
[+] find person feature
[+] PersonProcessor and PersonRepository specific interface for person specific methods
[...] Exception handlers moved to interface over controllers
Part2

TerKaCe committed:
[+] Person tests
[+] parallel tests run

Labyc committed:
[...] Service doesn't work with DTOs
[-] Optionals from method parameters
[...] Move test to appropriate package and class

TerKaCe committed:
[!] unworkable, move

TerKaCe committed:
[!] unworkable, move

TerKaCe committed:
[!] unworkable, move
